<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Data Preprocessing</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">RNA-Seq Data Analysis at OSC</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="introR.html">Intro to R</a>
</li>
<li>
  <a href="rna_seq_analysis.html">Day 1</a>
</li>
<li>
  <a href="rna_seq_analysis_day2.html">Day 2</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Data Preprocessing</h1>

</div>


<style type="text/css">
body{ /* Normal  */
      font-size: 16px;
  }

code.r{ /* Code block */
    font-size: 14px;
}
</style>
<div id="overview" class="section level2">
<h2><span class="header-section-number">0.1</span> Overview</h2>
<ul>
<li>Reading in table of counts</li>
<li>Filtering lowly expressed genes</li>
<li>Quality control</li>
<li>Normalization for composition bias</li>
</ul>
</div>
<div id="experimental-design" class="section level2">
<h2><span class="header-section-number">0.2</span> Experimental Design</h2>
<p>Data for this workshop comes from a Nature Scientific Data paper by <a href="https://www.nature.com/articles/sdata2018290">Zhang et al., 2018</a>. The study examimes mechanisms of saline adaptation of a plant (<em>Strophostyles helvola</em>) related to the common bean (<em>Phaseolus vulgaris</em>). Briefly stated, the study reported the RNA-seq analyses of two genotypes (a salt tolerant beach genotype and a salt-sensitive inland genotype, see figure bellow) and provided sequence data for community use. Sequence data was deposited into the National Center for Biotechnology Information (NCBI) Sequence Read Archive (SRA) repository, and the abundance count for all the samples was deposited at Gene Expression Omnibus (GEO) database.</p>
<p>
</p>
<div style="text-align: center; margin-top: 10px; margin-bottom: 10px;">
<p><img src="flowChart.png" /></p>
</div>
<p>
</p>
<p>First, we load required packages:</p>
<pre class="r"><code>library(edgeR) #Main package for DE analysis
library(tidyverse) #Data wrangling package, includes ggplot2, dplyr, tidy, readr
library(RColorBrewer) #Colour scheme for plotting
library(Glimma) #Interactive MD plots
library(gplots)</code></pre>
<p>For this workshop, the count data from the study will be obtained from the figshare repository, as shown bellow:</p>
<pre class="r"><code>download.file(url = &quot;https://ndownloader.figshare.com/files/23241812&quot;,
              destfile = &quot;gene_counts.txt&quot;)</code></pre>
<p>Import data into <code>R</code>:</p>
<pre class="r"><code>raw_counts &lt;- read.table(file=&quot;gene_counts.txt&quot;, 
                         sep = &quot;\t&quot;, 
                         header = TRUE
)</code></pre>
<p>This is a data frame with the first 6 columns describing features (genes). Samples are from column number 7 (SRR).</p>
<pre class="r"><code>head(raw_counts)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["Geneid"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Chr"],"name":[2],"type":["chr"],"align":["left"]},{"label":["Start"],"name":[3],"type":["int"],"align":["right"]},{"label":["End"],"name":[4],"type":["int"],"align":["right"]},{"label":["Strand"],"name":[5],"type":["chr"],"align":["left"]},{"label":["Length"],"name":[6],"type":["int"],"align":["right"]},{"label":["SRR7609467"],"name":[7],"type":["int"],"align":["right"]},{"label":["SRR7609468"],"name":[8],"type":["int"],"align":["right"]},{"label":["SRR7609469"],"name":[9],"type":["int"],"align":["right"]},{"label":["SRR7609470"],"name":[10],"type":["int"],"align":["right"]},{"label":["SRR7609471"],"name":[11],"type":["int"],"align":["right"]},{"label":["SRR7609472"],"name":[12],"type":["int"],"align":["right"]},{"label":["SRR7609473"],"name":[13],"type":["int"],"align":["right"]},{"label":["SRR7609474"],"name":[14],"type":["int"],"align":["right"]},{"label":["SRR7609475"],"name":[15],"type":["int"],"align":["right"]},{"label":["SRR7609476"],"name":[16],"type":["int"],"align":["right"]},{"label":["SRR7609477"],"name":[17],"type":["int"],"align":["right"]},{"label":["SRR7609478"],"name":[18],"type":["int"],"align":["right"]}],"data":[{"1":"Phvul.001G000100.v1.0","2":"Chr01","3":"3852","4":"11015","5":"-","6":"7164","7":"138","8":"123","9":"142","10":"154","11":"116","12":"128","13":"133","14":"132","15":"111","16":"91","17":"116","18":"147","_rn_":"1"},{"1":"Phvul.001G000200.v1.0","2":"Chr01","3":"99177","4":"104451","5":"-","6":"5275","7":"0","8":"0","9":"0","10":"0","11":"0","12":"0","13":"0","14":"0","15":"0","16":"0","17":"0","18":"0","_rn_":"2"},{"1":"Phvul.001G000300.v1.0","2":"Chr01","3":"125452","4":"132387","5":"+","6":"6936","7":"3458","8":"3530","9":"3742","10":"3979","11":"3339","12":"2912","13":"3597","14":"4190","15":"3329","16":"3433","17":"2652","18":"3121","_rn_":"3"},{"1":"Phvul.001G000400.v1.0","2":"Chr01","3":"134609","4":"140566","5":"-","6":"5958","7":"237","8":"258","9":"236","10":"259","11":"207","12":"216","13":"309","14":"252","15":"214","16":"208","17":"172","18":"221","_rn_":"4"},{"1":"Phvul.001G000500.v1.0","2":"Chr01","3":"144309","4":"146169","5":"+","6":"1861","7":"1624","8":"1351","9":"1650","10":"1955","11":"1551","12":"902","13":"1639","14":"2136","15":"1908","16":"1701","17":"621","18":"730","_rn_":"5"},{"1":"Phvul.001G000600.v1.0","2":"Chr01","3":"149887","4":"162657","5":"+","6":"12771","7":"329","8":"351","9":"363","10":"384","11":"404","12":"230","13":"406","14":"370","15":"411","16":"359","17":"185","18":"232","_rn_":"6"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>Change sample names from the SRR to reflect sample information, such that: <strong>IT</strong>: inland treated; <strong>IC</strong>: inland control; <strong>BT</strong>: beach treated; <strong>BC</strong>: beach control</p>
<pre class="r"><code>names(raw_counts)[7:18] &lt;- c(&quot;IT2&quot;, &quot;BC2&quot;, &quot;IC3&quot;, &quot;IT1&quot;, &quot;BC3&quot;, &quot;BT1&quot;, &quot;BC1&quot;, &quot;IT3&quot;, &quot;IC1&quot;, &quot;IC2&quot;, &quot;BT2&quot;, &quot;BT3&quot;)

raw_counts[1:6, 7:18]</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["IT2"],"name":[1],"type":["int"],"align":["right"]},{"label":["BC2"],"name":[2],"type":["int"],"align":["right"]},{"label":["IC3"],"name":[3],"type":["int"],"align":["right"]},{"label":["IT1"],"name":[4],"type":["int"],"align":["right"]},{"label":["BC3"],"name":[5],"type":["int"],"align":["right"]},{"label":["BT1"],"name":[6],"type":["int"],"align":["right"]},{"label":["BC1"],"name":[7],"type":["int"],"align":["right"]},{"label":["IT3"],"name":[8],"type":["int"],"align":["right"]},{"label":["IC1"],"name":[9],"type":["int"],"align":["right"]},{"label":["IC2"],"name":[10],"type":["int"],"align":["right"]},{"label":["BT2"],"name":[11],"type":["int"],"align":["right"]},{"label":["BT3"],"name":[12],"type":["int"],"align":["right"]}],"data":[{"1":"138","2":"123","3":"142","4":"154","5":"116","6":"128","7":"133","8":"132","9":"111","10":"91","11":"116","12":"147","_rn_":"1"},{"1":"0","2":"0","3":"0","4":"0","5":"0","6":"0","7":"0","8":"0","9":"0","10":"0","11":"0","12":"0","_rn_":"2"},{"1":"3458","2":"3530","3":"3742","4":"3979","5":"3339","6":"2912","7":"3597","8":"4190","9":"3329","10":"3433","11":"2652","12":"3121","_rn_":"3"},{"1":"237","2":"258","3":"236","4":"259","5":"207","6":"216","7":"309","8":"252","9":"214","10":"208","11":"172","12":"221","_rn_":"4"},{"1":"1624","2":"1351","3":"1650","4":"1955","5":"1551","6":"902","7":"1639","8":"2136","9":"1908","10":"1701","11":"621","12":"730","_rn_":"5"},{"1":"329","2":"351","3":"363","4":"384","5":"404","6":"230","7":"406","8":"370","9":"411","10":"359","11":"185","12":"232","_rn_":"6"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>Reorder columns to group treatment samples together, by location</p>
<pre class="r"><code>raw_counts &lt;- raw_counts[ , c(1:6, 9,15,16,7,10,14,8,11,13,12,17,18)]
head(raw_counts[,7:18])</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["IC3"],"name":[1],"type":["int"],"align":["right"]},{"label":["IC1"],"name":[2],"type":["int"],"align":["right"]},{"label":["IC2"],"name":[3],"type":["int"],"align":["right"]},{"label":["IT2"],"name":[4],"type":["int"],"align":["right"]},{"label":["IT1"],"name":[5],"type":["int"],"align":["right"]},{"label":["IT3"],"name":[6],"type":["int"],"align":["right"]},{"label":["BC2"],"name":[7],"type":["int"],"align":["right"]},{"label":["BC3"],"name":[8],"type":["int"],"align":["right"]},{"label":["BC1"],"name":[9],"type":["int"],"align":["right"]},{"label":["BT1"],"name":[10],"type":["int"],"align":["right"]},{"label":["BT2"],"name":[11],"type":["int"],"align":["right"]},{"label":["BT3"],"name":[12],"type":["int"],"align":["right"]}],"data":[{"1":"142","2":"111","3":"91","4":"138","5":"154","6":"132","7":"123","8":"116","9":"133","10":"128","11":"116","12":"147","_rn_":"1"},{"1":"0","2":"0","3":"0","4":"0","5":"0","6":"0","7":"0","8":"0","9":"0","10":"0","11":"0","12":"0","_rn_":"2"},{"1":"3742","2":"3329","3":"3433","4":"3458","5":"3979","6":"4190","7":"3530","8":"3339","9":"3597","10":"2912","11":"2652","12":"3121","_rn_":"3"},{"1":"236","2":"214","3":"208","4":"237","5":"259","6":"252","7":"258","8":"207","9":"309","10":"216","11":"172","12":"221","_rn_":"4"},{"1":"1650","2":"1908","3":"1701","4":"1624","5":"1955","6":"2136","7":"1351","8":"1551","9":"1639","10":"902","11":"621","12":"730","_rn_":"5"},{"1":"363","2":"411","3":"359","4":"329","5":"384","6":"370","7":"351","8":"404","9":"406","10":"230","11":"185","12":"232","_rn_":"6"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>Import the count data into the <code>edgeR</code> package. This will create a <code>DGEList</code> object for <code>edgeR</code>. The object has a number of slots for storing various parameters about the data.</p>
<pre class="r"><code>dge &lt;- DGEList(counts = raw_counts[ , 7:18],
               lib.size = colSums(raw_counts[ , 7:18]),
               norm.factors = rep(1,ncol(raw_counts[ , 7:18])),
               samples = NULL,
               group = NULL,
               genes = raw_counts[ , 1:6])</code></pre>
<p>View the different data types contained in the <code>DGEList</code> object:</p>
<pre class="r"><code>dge</code></pre>
<pre><code>## An object of class &quot;DGEList&quot;
## $counts
##    IC3  IC1  IC2  IT2  IT1  IT3  BC2  BC3  BC1  BT1  BT2  BT3
## 1  142  111   91  138  154  132  123  116  133  128  116  147
## 2    0    0    0    0    0    0    0    0    0    0    0    0
## 3 3742 3329 3433 3458 3979 4190 3530 3339 3597 2912 2652 3121
## 4  236  214  208  237  259  252  258  207  309  216  172  221
## 5 1650 1908 1701 1624 1955 2136 1351 1551 1639  902  621  730
## 27192 more rows ...
## 
## $samples
##     group lib.size norm.factors
## IC3     1 40619905            1
## IC1     1 35114368            1
## IC2     1 34833786            1
## IT2     1 36399307            1
## IT1     1 41939123            1
## 7 more rows ...
## 
## $genes
##                  Geneid   Chr  Start    End Strand Length
## 1 Phvul.001G000100.v1.0 Chr01   3852  11015      -   7164
## 2 Phvul.001G000200.v1.0 Chr01  99177 104451      -   5275
## 3 Phvul.001G000300.v1.0 Chr01 125452 132387      +   6936
## 4 Phvul.001G000400.v1.0 Chr01 134609 140566      -   5958
## 5 Phvul.001G000500.v1.0 Chr01 144309 146169      +   1861
## 27192 more rows ...</code></pre>
<p>Add sample and location information. Note that for a multivariate model, all combinations of treatments/group should be replicated in the second variable/factor. That is, we should have Control (C) and Treatment (T) samples, i.e group factor, in <strong>both</strong> inland and beach locations.</p>
<pre class="r"><code>#group factor:
group&lt;-as.factor(rep(c(&quot;C&quot;,&quot;T&quot;,&quot;C&quot;,&quot;T&quot;), c(3,3,3,3)))
dge$samples$group&lt;-group
group</code></pre>
<pre><code>##  [1] C C C T T T C C C T T T
## Levels: C T</code></pre>
<pre class="r"><code>#location factor:
location&lt;-as.factor(rep(c(&quot;inland&quot;,&quot;beach&quot;),c(6,6)))
dge$samples$location&lt;-location
location</code></pre>
<pre><code>##  [1] inland inland inland inland inland inland beach  beach  beach  beach  beach  beach 
## Levels: beach inland</code></pre>
<p>Next, we verify that we have “C” and “T” in both inland and beach:</p>
<pre class="r"><code>dge$samples</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["group"],"name":[1],"type":["fctr"],"align":["left"]},{"label":["lib.size"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["norm.factors"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["location"],"name":[4],"type":["fctr"],"align":["left"]}],"data":[{"1":"C","2":"40619905","3":"1","4":"inland","_rn_":"IC3"},{"1":"C","2":"35114368","3":"1","4":"inland","_rn_":"IC1"},{"1":"C","2":"34833786","3":"1","4":"inland","_rn_":"IC2"},{"1":"T","2":"36399307","3":"1","4":"inland","_rn_":"IT2"},{"1":"T","2":"41939123","3":"1","4":"inland","_rn_":"IT1"},{"1":"T","2":"42597147","3":"1","4":"inland","_rn_":"IT3"},{"1":"C","2":"32496003","3":"1","4":"beach","_rn_":"BC2"},{"1":"C","2":"31262960","3":"1","4":"beach","_rn_":"BC3"},{"1":"C","2":"33240392","3":"1","4":"beach","_rn_":"BC1"},{"1":"T","2":"31682364","3":"1","4":"beach","_rn_":"BT1"},{"1":"T","2":"30170320","3":"1","4":"beach","_rn_":"BT2"},{"1":"T","2":"34386767","3":"1","4":"beach","_rn_":"BT3"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>Save the <code>DGEList</code> object in order to maintain an original copy.</p>
<pre class="r"><code>dge_orig&lt;-dge
saveRDS(dge_orig, file = &quot;dge_orig.rds&quot;)</code></pre>
</div>
<div id="cpm-transformation" class="section level2">
<h2><span class="header-section-number">0.3</span> CPM transformation</h2>
<p>We will transform raw read counts to expression counts: counts per million (CPM). While the majority of transformation methods work well, RPKM and FPKM normalization should be avoided in the context of differential expression analysis, no matter how often you see them applied in published studies. RPKM and FPKM are only needed if expression values need to be compared between different genes within the same sample for which the different gene lengths must be taken into consideration.</p>
<p>We use the <code>cpm</code> function for transforming raw counts.</p>
<pre class="r"><code>#raw counts are converted to CPM and log-CPM values using the cpm function
cpm &lt;- cpm(dge) 
lcpm &lt;- cpm(dge, log=TRUE)

L &lt;- mean(dge$samples$lib.size) * 1e-6 #average library size in Millions
M &lt;- median(dge$samples$lib.size) * 1e-6 #median lib size </code></pre>
<p>View the distribution of log-CPM, for inland control replicates:</p>
<pre class="r"><code>#summary(lcpm)
summary(cpm)[,1:3]</code></pre>
<pre><code>##       IC3                 IC1                 IC2           
##  Min.   :    0.000   Min.   :    0.000   Min.   :    0.000  
##  1st Qu.:    0.517   1st Qu.:    0.513   1st Qu.:    0.517  
##  Median :    9.183   Median :    9.512   Median :    9.244  
##  Mean   :   36.769   Mean   :   36.769   Mean   :   36.769  
##  3rd Qu.:   35.894   3rd Qu.:   36.737   3rd Qu.:   35.598  
##  Max.   :14527.705   Max.   :16703.447   Max.   :14496.414</code></pre>
</div>
<div id="gene-filtering" class="section level2">
<h2><span class="header-section-number">0.4</span> Gene filtering</h2>
<p>Genes with very low counts across all libraries provide little evidence for differential expression and they interfere with some of the statistical approximations that are used later in the pipeline. They also add to the multiple testing burden when estimating false discovery rates, reducing power to detect DEGs. These genes should be filtered out prior to further analysis.</p>
<p>We remove genes that are lowly expressed using the <code>filterByExpr</code> edgeR function. By default, the function keeps genes with about 10 read counts or more in a minimum number of samples, where the number of samples is chosen according to the minimum group sample size. The actual filtering uses CPM values rather than counts in order to avoid giving preference to samples with large library sizes. For example, if the median library size is about 51 million and 10/51 ≈ 0.2, so the <code>filterByExpr</code> function keeps genes that have a CPM of 0.2 or more in at least three samples.</p>
<pre class="r"><code>dim(dge) #retrieve dimentions of the dge object</code></pre>
<pre><code>## [1] 27197    12</code></pre>
<pre class="r"><code>keep.exprs &lt;- filterByExpr(dge)
#Minimum read count can also be explicitly specified, for instance &#39;5&#39; bellow: 
#keep.exprs &lt;- filterByExpr(dge, min.count = 5, min.total.count = 5)

dge &lt;- dge[keep.exprs,, keep.lib.sizes=FALSE]
dim(dge)</code></pre>
<pre><code>## [1] 21181    12</code></pre>
<p>Let us plot line graphs depicting the effect of filtering out genes with low expression values:</p>
<pre class="r"><code>lcpm.cutoff &lt;- log2(10/M + 2/L)
nsamples &lt;- ncol(dge)
col &lt;- brewer.pal(nsamples, &quot;Paired&quot;)
par(mfrow=c(1,2))
plot(density(lcpm[,1]), col=col[1], lwd=2, ylim=c(0,0.26), las=2, main=&quot;&quot;, xlab=&quot;&quot;) #density with lcpm from unfiltered data
title(main=&quot;A. Raw data&quot;, xlab=&quot;Log-cpm&quot;)
abline(v=lcpm.cutoff, lty=3)
for (i in 2:nsamples){
  den &lt;- density(lcpm[,i])
  lines(den$x, den$y, col=col[i], lwd=2)
}
#legend(&quot;topright&quot;, samplenames, text.col=col, bty=&quot;n&quot;)
lcpm2 &lt;- cpm(dge, log=TRUE) #NOTE: recalculating lcpm from filtered data!
plot(density(lcpm2[,1]), col=col[1], lwd=2, ylim=c(0,0.26), las=2, main=&quot;&quot;, xlab=&quot;&quot;)
title(main=&quot;B. Filtered data&quot;, xlab=&quot;Log-cpm&quot;)
abline(v=lcpm.cutoff, lty=3)
for (i in 2:nsamples){
  den &lt;- density(lcpm2[,i])
  lines(den$x, den$y, col=col[i], lwd=2)
}</code></pre>
<p><img src="rna_seq_analysis_files/figure-html/unnamed-chunk-16-1.png" width="672" /></p>
<pre class="r"><code>#legend(&quot;topright&quot;, samplenames, text.col=col, bty=&quot;n&quot;)</code></pre>
</div>
<div id="normalization-for-sequencing-depth-differences" class="section level2">
<h2><span class="header-section-number">0.5</span> Normalization for sequencing depth differences</h2>
<p>Normalization is a process designed to identify and remove systematic technical differences between samples that occur in the data to ensure that technical bias has minimal impact on the results.</p>
<p>Bar plots bellow show need for normalization to account for different library sizes.</p>
<pre class="r"><code>par(mfrow=c(1,1))
samplenames&lt;-c(&quot;Inland_c3&quot;,&quot;Inland_c1&quot;,&quot;Inland_c2&quot;,&quot;Inland_t2&quot;,&quot;Inland_t1&quot;,&quot;Inland_t3&quot;,
               &quot;Beach_c2&quot;,&quot;Beach_c3&quot;,&quot;Beach_c1&quot;,&quot;Beach_t1&quot;,&quot;Beach_t2&quot;,&quot;Beach_t3&quot;)
#Lib sizes:
barplot(dge$samples$lib.size, las = 2, names.arg = samplenames)</code></pre>
<p><img src="rna_seq_analysis_files/figure-html/unnamed-chunk-17-1.png" width="672" /></p>
<p>The overall strategy with normalization is to choose an appropriate baseline, and express sample counts relative to that baseline.</p>
<p>At lot of different normalization methods exist, but we will use TMM normalization in this tutorial.</p>
<p>
</p>
<div style="text-align: center; margin-top: 10px; margin-bottom: 10px;">
<p><img src="normalization.png" /></p>
</div>
<p>
</p>
<p>Developed by <a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2010-11-3-r25">Mark D Robinson and Oshlack, 2010</a>, the trimmed mean of M-values normalization method (TMM) is performed to eliminate composition biases between libraries. This generates a set of normalization factors, where the product of these factors and the library sizes defines the effective library size. The <code>calcNormFactors</code> function in edgeR calculates the normalization factors between libraries.</p>
<p>Below, we first create a copy of raw data, followed by TMM normalization:</p>
<pre class="r"><code>dge_unNorm&lt;-dge 
dge &lt;- calcNormFactors(dge, method = &quot;TMM&quot;)</code></pre>
<p>Note the change in normalization factors:</p>
<pre class="r"><code>dge$samples</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["group"],"name":[1],"type":["fctr"],"align":["left"]},{"label":["lib.size"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["norm.factors"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["location"],"name":[4],"type":["fctr"],"align":["left"]}],"data":[{"1":"C","2":"40607955","3":"0.9620536","4":"inland","_rn_":"IC3"},{"1":"C","2":"35103736","3":"0.9960395","4":"inland","_rn_":"IC1"},{"1":"C","2":"34823466","3":"0.9552281","4":"inland","_rn_":"IC2"},{"1":"T","2":"36387017","3":"1.0302307","4":"inland","_rn_":"IT2"},{"1":"T","2":"41927438","3":"1.0302259","4":"inland","_rn_":"IT1"},{"1":"T","2":"42584763","3":"1.0358925","4":"inland","_rn_":"IT3"},{"1":"C","2":"32486368","3":"1.0088472","4":"beach","_rn_":"BC2"},{"1":"C","2":"31254455","3":"1.0095501","4":"beach","_rn_":"BC3"},{"1":"C","2":"33228637","3":"1.0391711","4":"beach","_rn_":"BC1"},{"1":"T","2":"31670175","3":"0.9896106","4":"beach","_rn_":"BT1"},{"1":"T","2":"30157075","3":"0.9586289","4":"beach","_rn_":"BT2"},{"1":"T","2":"34373388","3":"0.9896469","4":"beach","_rn_":"BT3"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>Let us plot the distribution of expression values showing the effect of normalization</p>
<pre class="r"><code>par(mfrow=c(1,2)) #create two panels for plotting

lcpm &lt;- cpm(dge_unNorm, log=TRUE)
boxplot(lcpm, las=2, col=col, main=&quot;&quot;)
title(main=&quot;Unnormalized data&quot;,ylab=&quot;Log-cpm&quot;)

lcpm &lt;- cpm(dge, log=TRUE)
boxplot(lcpm, las=2, col=col, main=&quot;&quot;)
title(main=&quot;Normalized data&quot;,ylab=&quot;Log-cpm&quot;)</code></pre>
<p><img src="rna_seq_analysis_files/figure-html/unnamed-chunk-20-1.png" width="672" /></p>
</div>
<div id="multidimensional-scaling-mds-plots" class="section level2">
<h2><span class="header-section-number">0.6</span> Multidimensional scaling (MDS) plots</h2>
<p>By far, one of the most important plots we make when we analyze RNA-Seq data are MDSplots. An MDSplot is a visualization of a principle components analysis, which determines the greatest sources of variation in the data. A principle components analysis is an example of an unsupervised analysis, where we don’t need to specify the groups. If your experiment is well controlled and has worked well, what we hope to see is that the greatest sources of variation in the data are the treatments/groups we are interested in. It is also an incredibly useful tool for quality control and checking for outliers. We can use the <code>plotMDS</code> function to create the MDS plot.</p>
<pre class="r"><code>lcpm &lt;- cpm(dge, log=TRUE)

#assign different colors to the group information
col.group &lt;- dge$samples$group
levels(col.group) &lt;- brewer.pal(nlevels(col.group), &quot;Set1&quot;)</code></pre>
<pre><code>## Warning in brewer.pal(nlevels(col.group), &quot;Set1&quot;): minimal value for n is 3, returning requested palette with 3 different levels</code></pre>
<pre class="r"><code>col.group &lt;- as.character(col.group)

par(mfrow=c(1,1))
plotMDS(lcpm, col=col.group) </code></pre>
<p><img src="rna_seq_analysis_files/figure-html/unnamed-chunk-21-1.png" width="672" /></p>
<p>Note from the plot above that all replicates per condition (treatment, location) cluster together.</p>
<blockquote>
<p><strong>Learning check!</strong></p>
<ul>
<li>What is the greatest source of variation in the data (i.e. what does dimension 1 represent)?</li>
<li>What is the second greatest source of variation in the data?</li>
</ul>
</blockquote>
<p>We observe from the figure above that greatest source of variation is the treatment effect (dimension 1). Also, and most importantly, note the separation due to location (dimension 2). This is suggestive that there might be need to account for location effect in the regression model. In other words, we can fit a model that determines the effect of treatment while adjusting for any differences in expression due to the effect of location. This is a type of multivariate or multi-factor analysis, and will be covered in session 2 of the worshop.</p>
<p>Next, we determine and plot out the proportion of explained variance for each Dimension observed in the MDSplot. We first transpose the data to have variables (genes) as columns:</p>
<pre class="r"><code>data_for_PCA &lt;- t(dge$counts)
dim(data_for_PCA)</code></pre>
<pre><code>## [1]    12 21181</code></pre>
<p>We then calculate matrix of dissimilarities between samples using the <code>dist</code> function, then calculate the classical multidimensional scaling of the matrix data using the <code>cmdsclale</code> function:</p>
<pre class="r"><code># Below, &quot;k&quot; is the maximum dimension of the space which the data are to be represented in
mds &lt;- cmdscale(dist(data_for_PCA), k=3, eig=TRUE)  

# transform the Eigen values into percentage
eig_pc &lt;- mds$eig * 100 / sum(mds$eig)

# barplot of the the percentage Eigen values
barplot(eig_pc,
        las=1,
        xlab=&quot;Dimension&quot;, 
        ylab=&quot;Proportion of explained variance (%)&quot;, y.axis=NULL,
        col=&quot;darkgrey&quot;, names.arg = 1:12)</code></pre>
<p><img src="rna_seq_analysis_files/figure-html/unnamed-chunk-23-1.png" width="672" /></p>
<p>As expected, dimension 1 accounts for the majority of variance, i.e treatment effect seem to drive the majority of differential expression.</p>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
