<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Multi-factor RNA-Seq Data Analysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">RNA-Seq Data Analysis at OSC</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="introR.html">Intro to R</a>
</li>
<li>
  <a href="rna_seq_analysis.html">Day 1</a>
</li>
<li>
  <a href="rna_seq_analysis_day2.html">Day 2</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Multi-factor RNA-Seq Data Analysis</h1>

</div>


<div id="model-fitting-and-hypothesis-testing" class="section level2">
<h2><span class="header-section-number">0.1</span> Model fitting and Hypothesis testing</h2>
<div id="method-2-accounting-for-location" class="section level3">
<h3><span class="header-section-number">0.1.1</span> Method 2: Accounting for location</h3>
<p>Recall that from Method 1, we fit two separate models on data from the two locations, and identifies DEGs due to application of salt separately. If the biological question is to identify the effect of treatment while the location is held contstant (i.e the effect of treatment irrespective of the location), then we need to account for location. We observed from MDSplot from Day 1 that there’s some (biological) variation explained by location due to separation of samples by Dimension 2. We therefore need to fit a multi-factor (multi-variate) additive model that accounts for location as a covariate.</p>
<p>Let us first load required packages:</p>
<pre class="r"><code>library(edgeR) #Main package for DE analysis
library(tidyverse) #Data wrangling package, includes ggplot2, dplyr, tidy, readr
library(RColorBrewer) #Colour scheme for plotting
library(Glimma) #Interactive MD plots
library(gplots)</code></pre>
<p>Then get the original copy of the DGE object</p>
<pre class="r"><code>dge_method2&lt;-readRDS(&quot;dge_orig.rds&quot;)</code></pre>
<p>Note that for a multivariate model, all combinations of treatments/group should be replicated in the second variable/factor. That is, we should have Control (C) and Treatment (T) samples in <strong>both</strong> inland and beach. We specify the group factor in the <code>dge_method2</code> to reflect this:</p>
<pre class="r"><code>group&lt;-as.factor(rep(c(&quot;C&quot;,&quot;T&quot;,&quot;C&quot;,&quot;T&quot;), c(3,3,3,3)))
dge_method2$samples$group&lt;-group
#verify that we have &quot;C&quot; and &quot;T&quot; in both inland and beach
dge_method2$samples</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["group"],"name":[1],"type":["fctr"],"align":["left"]},{"label":["lib.size"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["norm.factors"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["location"],"name":[4],"type":["fctr"],"align":["left"]}],"data":[{"1":"C","2":"40619905","3":"1","4":"inland","_rn_":"IC3"},{"1":"C","2":"35114368","3":"1","4":"inland","_rn_":"IC1"},{"1":"C","2":"34833786","3":"1","4":"inland","_rn_":"IC2"},{"1":"T","2":"36399307","3":"1","4":"inland","_rn_":"IT2"},{"1":"T","2":"41939123","3":"1","4":"inland","_rn_":"IT1"},{"1":"T","2":"42597147","3":"1","4":"inland","_rn_":"IT3"},{"1":"C","2":"32496003","3":"1","4":"beach","_rn_":"BC2"},{"1":"C","2":"31262960","3":"1","4":"beach","_rn_":"BC3"},{"1":"C","2":"33240392","3":"1","4":"beach","_rn_":"BC1"},{"1":"T","2":"31682364","3":"1","4":"beach","_rn_":"BT1"},{"1":"T","2":"30170320","3":"1","4":"beach","_rn_":"BT2"},{"1":"T","2":"34386767","3":"1","4":"beach","_rn_":"BT3"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<div id="data-preprocessing" class="section level4">
<h4><span class="header-section-number">0.1.1.1</span> Data preprocessing</h4>
<p>First, let us preprocess the count data contained in the <code>dge_methods2</code> object</p>
<pre class="r"><code>  # perform cpm and lcp transformation:
  cpm &lt;- cpm(dge_method2)
  lcpm &lt;- cpm(dge_method2, log=TRUE)
  #filter low expressed genes
  keep.exprs &lt;- filterByExpr(dge_method2)
  dge_method2 &lt;- dge_method2[keep.exprs,, keep.lib.sizes=FALSE]
  ##TMM normalization:
  dge_method2 &lt;- calcNormFactors(dge_method2, method = &quot;TMM&quot;)</code></pre>
</div>
<div id="design-matrix" class="section level4">
<h4><span class="header-section-number">0.1.1.2</span> Design matrix</h4>
<p>Let us create a design matrix, this time round <strong>with</strong> intercept for the location variable. Also,this type of additive model corrects/adjusts for baseline differences between the locations. <strong>IMPORTANT:</strong> The order of factors matters—always put the factor of interest last!</p>
<pre class="r"><code>location &lt;- dge_method2$samples$location    
mod &lt;- model.matrix(~ location + group)
rownames(mod) &lt;- colnames(dge_method2)</code></pre>
<p>Let us take a look at the new model:</p>
<pre class="r"><code>mod</code></pre>
<pre><code>##     (Intercept) locationinland groupT
## IC3           1              1      0
## IC1           1              1      0
## IC2           1              1      0
## IT2           1              1      1
## IT1           1              1      1
## IT3           1              1      1
## BC2           1              0      0
## BC3           1              0      0
## BC1           1              0      0
## BT1           1              0      1
## BT2           1              0      1
## BT3           1              0      1
## attr(,&quot;assign&quot;)
## [1] 0 1 2
## attr(,&quot;contrasts&quot;)
## attr(,&quot;contrasts&quot;)$location
## [1] &quot;contr.treatment&quot;
## 
## attr(,&quot;contrasts&quot;)$group
## [1] &quot;contr.treatment&quot;</code></pre>
<p>We see three coefficients (sometimes called “dummy variables”), the last coefficient (coefficient 3) representing the effect of treatment on gene expression, after adjusting for the location. This is our coefficient of interest.</p>
<p>Next we estimate dispersion:</p>
<pre class="r"><code>  dge_method2 &lt;- estimateDisp(dge_method2, design = mod)</code></pre>
<p>We fit a GLM</p>
<pre class="r"><code>  fit &lt;- glmFit(dge2, mod)</code></pre>
<p>We perform hypothesis testing to answer the following biological question: * What genes are differentally expressed under treatment, after accounting for location? Note below that we do not build a contrast matrix, but rather we specify the coefficient of interest (3) while performing the LRT.</p>
<pre class="r"><code>#LRT
glf_T_vs_C &lt;-glmLRT(fit, coef = 3)

#perform multiple test correction, and obtain DEGs with FDR cutfoff of 0.1
pvals_T_vs_C &lt;- topTags(glf_T_vs_C, n = &quot;Inf&quot;, adjust.method = &quot;BH&quot;, sort.by = &quot;PValue&quot;, p.value = 0.1)[[1]]

head(pvals_T_vs_C)</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["Geneid"],"name":[1],"type":["chr"],"align":["left"]},{"label":["Chr"],"name":[2],"type":["chr"],"align":["left"]},{"label":["Start"],"name":[3],"type":["int"],"align":["right"]},{"label":["End"],"name":[4],"type":["int"],"align":["right"]},{"label":["Strand"],"name":[5],"type":["chr"],"align":["left"]},{"label":["Length"],"name":[6],"type":["int"],"align":["right"]},{"label":["logFC"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["logCPM"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["LR"],"name":[9],"type":["dbl"],"align":["right"]},{"label":["PValue"],"name":[10],"type":["dbl"],"align":["right"]},{"label":["FDR"],"name":[11],"type":["dbl"],"align":["right"]}],"data":[{"1":"Phvul.004G056400.v1.0","2":"Chr04","3":"7396606","4":"7397508","5":"-","6":"903","7":"6.107241","8":"4.368390","9":"1255.6940","10":"4.804607e-275","11":"1.040918e-270","_rn_":"9569"},{"1":"Phvul.001G263200.v1.0","2":"Chr01","3":"51729773","4":"51729982","5":"+","6":"210","7":"5.094781","8":"4.083698","9":"1198.6100","10":"1.222857e-262","11":"1.324660e-258","_rn_":"2632"},{"1":"Phvul.001G075400.v1.0","2":"Chr01","3":"10506529","4":"10509049","5":"-","6":"2521","7":"3.981461","8":"6.171969","9":"941.1945","10":"1.088544e-206","11":"7.861104e-203","_rn_":"754"},{"1":"Phvul.007G231800.v1.0","2":"Chr07","3":"47146783","4":"47148189","5":"-","6":"1407","7":"5.104610","8":"6.635312","9":"920.7311","10":"3.056293e-202","11":"1.655365e-198","_rn_":"17196"},{"1":"Phvul.007G100900.v1.0","2":"Chr07","3":"10966196","4":"10970519","5":"-","6":"4324","7":"2.482682","8":"5.402644","9":"911.2133","10":"3.582612e-200","11":"1.552346e-196","_rn_":"15887"},{"1":"Phvul.001G067300.v1.0","2":"Chr01","3":"8642080","4":"8646688","5":"-","6":"4609","7":"4.833595","8":"6.778686","9":"894.5400","10":"1.509270e-196","11":"5.449722e-193","_rn_":"673"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<blockquote>
<p><strong>Learning check!</strong></p>
<ul>
<li>How many genes are differentially expressed?</li>
</ul>
</blockquote>
</div>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
