<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Wilbur Ouma" />


<title>RNA-Seq Data Analysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">RNA-Seq Data Analysis at OSC</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">Before we start</a>
</li>
<li>
  <a href="rna_seq_analysis.html">Day 1</a>
</li>
<li>
  <a href="rna_seq_analysis_day2.html">Day 2</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">RNA-Seq Data Analysis</h1>
<h4 class="author">Wilbur Ouma</h4>

</div>


<p>Load required packages:</p>
<pre class="r"><code>library(edgeR) #Main package for DE analysis
library(tidyverse) #Data wrangling package, includes ggplot2, dplyr, tidy, readr
library(RColorBrewer) #Colour scheme for plotting
library(Glimma) #Interactive MD plots
library(gplots)</code></pre>
<div id="method-2" class="section level3">
<h3><span class="header-section-number">0.0.1</span> Method 2</h3>
<p>RECALL: MDS plot (fig. ?), there’s some (biological) variation explained by location, but.. remember we combined all replicates of a location into one grp? Not a good idea, batch effect</p>
<p>Problem: variation acrross different locations not accounted for!</p>
<p>Account for gene variations across different location Why include location/batch factor? increase the sensitivity for finding differences due to treatment!!! Multivariate model, with location as a covariate treatments can be adjusted for differences between the batches by using an additive model formula of the form</p>
<p>Get the original copy of the DGE object</p>
<pre class="r"><code>dge_opt2&lt;-dge_orig</code></pre>
<p>Modify the $samples object:</p>
<pre class="r"><code>treatment&lt;-as.factor(rep(c(&quot;C&quot;,&quot;T&quot;,&quot;C&quot;,&quot;T&quot;), c(3,3,3,3)))
dge_opt2$samples$group&lt;-treatment
dge_opt2$samples</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["group"],"name":[1],"type":["fctr"],"align":["left"]},{"label":["lib.size"],"name":[2],"type":["dbl"],"align":["right"]},{"label":["norm.factors"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["location"],"name":[4],"type":["fctr"],"align":["left"]}],"data":[{"1":"C","2":"40619905","3":"1","4":"inland","_rn_":"IC3"},{"1":"C","2":"35114368","3":"1","4":"inland","_rn_":"IC1"},{"1":"C","2":"34833786","3":"1","4":"inland","_rn_":"IC2"},{"1":"T","2":"36399307","3":"1","4":"inland","_rn_":"IT2"},{"1":"T","2":"41939123","3":"1","4":"inland","_rn_":"IT1"},{"1":"T","2":"42597147","3":"1","4":"inland","_rn_":"IT3"},{"1":"C","2":"32496003","3":"1","4":"beach","_rn_":"BC2"},{"1":"C","2":"31262960","3":"1","4":"beach","_rn_":"BC3"},{"1":"C","2":"33240392","3":"1","4":"beach","_rn_":"BC1"},{"1":"T","2":"31682364","3":"1","4":"beach","_rn_":"BT1"},{"1":"T","2":"30170320","3":"1","4":"beach","_rn_":"BT2"},{"1":"T","2":"34386767","3":"1","4":"beach","_rn_":"BT3"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>LEARNING CHECK: Obtain cpm values, filter out non-DEGs, perform (TMM) normalization; prepare MDS plots SOLUTION: R FUNCTIONS!!!! Example</p>
<pre class="r"><code>average&lt;-function(x){
  return(sum(x)/length(x))
}
average(c(5,6,7))</code></pre>
<pre><code>## [1] 6</code></pre>
<p>R function: Takes a DGEList edgeR object; performs the following: 1. Pre-processing (cpm, filter, normalization) 2. Fits a glm on data, get DEGs with LRT, for the separate locations</p>
<p>Note: Location NOT included as a covariate in the model, for this see OPTION 3</p>
<pre class="r"><code>degs_func_opt3&lt;-function(dge){
  #cpm:
  cpm &lt;- cpm(dge)
  lcpm &lt;- cpm(dge, log=TRUE)
  #filter
  keep.exprs &lt;- filterByExpr(dge)
  dge &lt;- dge[keep.exprs,, keep.lib.sizes=FALSE]
  ##Normalize:
  dge &lt;- calcNormFactors(dge, method = &quot;TMM&quot;)
  
  #Method 2:
    
  #treatments can be adjusted for differences between the batches (locations) by using 
  #an additive model formula of the form:
  location&lt;-dge$samples$location
  treatment&lt;-dge$samples$group
  #mod &lt;- model.matrix(~dge$samples$location + dge$samples$group) #the order matters...always put the factor of interest last! 
  mod &lt;- model.matrix(~location + treatment)
  #No intercept for location only!
  #In this type of analysis, the treatments are compared only within each location. The analysis is
  #corrected for baseline differences between the locations.
  #e treatments adjusting for any baseline differences between the
  #compare the two treatments adjusting for any baseline differences between the
  #locations by fitting an additive model
  #Note: no intercept (base-line) in the model
  
  rownames(mod) &lt;- colnames(dge)
  #colnames(mod)&lt;-c(&quot;BC&quot;,&quot;BT&quot;,&quot;IC&quot;,&quot;IT&quot;)
  mod
  
  #Estimate dispersion: 
  dge2 &lt;- estimateDisp(dge, design = mod)
  
  
  #Fit GLM or GLM QL:
  fit &lt;- glmFit(dge2, mod)
  
  #First we check whether there was a genuine need to adjust for location:
  qlf &lt;- glmLRT(fit, coef=2)
  topTags(qlf)
  FDR &lt;- p.adjust(qlf$table$PValue, method=&quot;BH&quot;)
  sum(FDR &lt;= 0.1) #8739genes. There is considerable differential expression, justifying our decision to adjust for the batch effect:
  ##LEARNING CHECK: Comppare this numver of DEGs with Option 1 Analysis!!!!!
  
  glf_T_vs_C &lt;-glmLRT(fit, coef = 3)
  
  
  #perform multiple testing correction/p value adjustment.
  #Why? Here: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6099145/
  
  
  
  pvals_T_vs_C &lt;- topTags(glf_T_vs_C, n = &quot;Inf&quot;, adjust.method = &quot;BH&quot;, sort.by = &quot;PValue&quot;, p.value = 0.1)[[1]]
  
  
  #return a dataframe with the counts of DEGs:
  deg &lt;- pvals_T_vs_C %&gt;%
    count() #11393 DEGs

  names(deg)&lt;-c(&quot;T_vs_C&quot;)
  row.names(deg)&lt;-&quot;# of DEGs&quot;
  return(deg)
  
}</code></pre>
<p>We can save the above the function in a text file (DEGsFunctionOption3.r), run it as shown bellow:</p>
<pre class="r"><code>source(&quot;../DEGsFunctionOption3.r&quot;, local = knitr::knit_global())
deg_opt2&lt;-degs_func_opt3(dge_opt2)
deg_opt2</code></pre>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["T_vs_C"],"name":[1],"type":["int"],"align":["right"]}],"data":[{"1":"11393","_rn_":"# of DEGs"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
